flink-job/src/main/java/com/ververica/field/dynamicrules/sources/TransactionsSource.java

```diff
@@ -61,9 +61,9 @@
   public static DataStream<Transaction> stringsStreamToTransactions(
       DataStream<String> transactionStrings) {
     return transactionStrings
-        .flatMap(new JsonDeserializer<Transaction>(Transaction.class))
+        .flatMap(new JsonDeserializer<>(Transaction.class))
```

Comment:
Nico, did you also verify that it works as expected? I vaguely remember having to leave the full signature in certain places, because otherwise the type system could not determine the types correctly (not sure if `returns()` helped).

Reply:
The job is running with that change. the diamond operator should imho work everywhere since it just makes the Java compiler deduct the type